tests/ui/codegen/equal-pointers-unequal/README.md

```diff
@@ -0,0 +1,22 @@
+See https://github.com/rust-lang/rust/issues/107975
+
+Basically, if you have two pointers with the same address but from two different allocations,
+the compiler gets confused whether their addresses are equal or not,
+resulting in some self-contradictory behavior of the compiled code.
+
+This folder contains some examples.
+They all boil down to allocating a variable on the stack, taking its address,
+getting rid of the variable, and then doing it all again.
+This way we end up with two addresses stored in two `usize`s (`a` and `b`).
+The addresses are (probably) equal but (definitely) come from two different allocations.
+Logically, we would expect that exactly one of the following options holds true:
+1. `a == b`
+2. `a != b`
+Sadly, the compiler does not always agree.
```

Comment:
It would be better to use a code example here, otherwise it will be difficult for people who don't know the issue to understand.

Reply:
A code example regarding the comparison, the concept of unequal pointers with equal addresses, or all of the above (something like what is already present [in the first test](https://github.com/rust-lang/rust/pull/127003/files/562aaa4be50ad0c649900b3c2fb8c9ede8d3c0d2#diff-b0b6598e3103d5a7b4bbd8475566e092ce54b1a2e6dc5bbf29c1de9fb5bee5e0))?

Reply:
A long description of program behavior may not be as friendly to the reader as a few lines of code. It is not a glaring problem though :)